src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -327,22 +334,56 @@ private void fileAclOp(AclOp op, Dataset dataset, List<String> fileIds, Map<IamR
     }
 
     /**
-     * Extract the bucket from a gs path.
+     * Extract the path portion (everything after the bucket name and it's trailing slash) of a gs path.
      */
-    public static String extractBucketFromPath(final String path) {
-        if (!path.startsWith("gs://") && StringUtils.countMatches(path, ',') < 3) {
-            throw new PdaoException(String.format("Bad GS path %s", path));
-        }
-        // Given a string that looks like "gs://some bucket/pat1/path2" (format verified above), this will extract
-        // "some bucket" and return
-        return path.substring(5).split("/")[0];
+    private static String extractFilePathInBucket(final String path, final String bucketName) {
+        return StringUtils.removeStart(path, String.format("gs://%s/", bucketName));
     }
 
     /**
-     * Extract the path portion (everything after the bucket name and it's trailing slash) of a gs path.
+     * Represents a way to access objects in GCS buckets.
      */
-    public static String extractFilePathInBucket(final String path,
-                                                 final String bucketName) {
-        return StringUtils.removeStart(path, String.format("gs://%s/", bucketName));
+    public static class GcsLocator {
+        private final String bucket;
```

Comment:
For struct-like object like this, my preference is to make the fields public and not bother with getters. I also wouldn't bother either equals and hashCode unless you know someone will be using it in a collection. (In JDK 14+, we can use `record` for these kind of objects)